### DISCUSSION TITLE
Unable to auto rotate Azure vault secrets with AKS

### DISCUSSION THREAD

**Discussion by gowthambalram** [Azure/secrets-store-csi-driver-provider-azure#903] [Q&A]
Hi,

We enabled auto syncSecret.enabled and enableSecretRotation at our AKS cluster and during the auto sync, driver pod is throwing the following error :

E0530 17:48:52.842825       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.22.0/tools/cache/reflector.go:167: Failed to watch *v1.Secret: failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:csi-secrets-store:secrets-store-csi-driver" cannot list resource "secrets" in API group "" at the cluster scope

Is there a way we can change service account to a different service account if its the issue with access to service account ?

**Comment by nilekhc** [comment]
@gowthambalram  Are you using AKS add-on? If so there was a bug related to missing Roles and Role Bindings. But It's fixed now.

Ref : https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/871
